### PR TITLE
Azure sources should all require unaltered access tokens

### PR DESCRIPTION
**Pull request by willmarks**
## Description

Azure sources should require unaltered access tokens

## Type of change
- [ ] Bug fix
- [X] New feature
- [X] Breaking change

### CODE REVIEW

File: sources/azure-repos-source/src/azure-repos.ts [hunk 1]
```diff
@@ -82,26 +87,26 @@ export class AzureRepos {
       throw new VError('Projects provided in addition to * keyword');
     }
 
-    const base64EncodedAccessToken = Buffer.from(
-      `${':'}${config.access_token}`,
-      'binary'
-    ).toString('base64');
+    let accessToken = config.access_token;
+    if (!isBase64Encoded(accessToken)) {
+      accessToken = base64Encode(`:${accessToken}`);
```
Comment (tovbinm) [inline sources/azure-repos-source/src/azure-repos.ts:92]: why not move the `:` to where you create the token

Comment (tovbinm) [inline sources/azure-repos-source/src/azure-repos.ts:92]: why not move the : to where you create the header?

Comment (willmarks) [reply sources/azure-repos-source/src/azure-repos.ts:92]: It needs to be part of the base64 encoded string.

Comment (tovbinm) [reply sources/azure-repos-source/src/azure-repos.ts:92]: oh, whoah, sucks!

Comment (tovbinm) [reply sources/azure-repos-source/src/azure-repos.ts:92]: how people are gonna even use it?

**Review by tovbinm — APPROVED** [review]

**Pull request merged**